src/generator/models.ts

```diff
@@ -58,6 +58,7 @@ export async function generateModels(session: Session<CodeModel>): Promise<strin
   }
   if (needsJSONPopulate) {
     text += 'func populate(m map[string]interface{}, k string, v interface{}) {\n';
+    text += '\tif v == nil {\n\t\treturn\n\t}\n';
     text += '\tif azcore.IsNullValue(v) {\n';
     text += '\t\tm[k] = nil\n';
     text += '\t} else if !reflect.ValueOf(v).IsNil() {\n';
```

Comment:
Also, is this else inside of the `if v== nil` check?

Reply:
No, the `v == nil` check is first.  I probably could condense it like so.

```go
if v == nil {
  return
} else if azcore.IsNullValue(v) {
  tm[k] = nil
} else if !reflect.ValueOf(v).IsNil() {
  tm[k] = v
}
```